src/persistence/src/persistent_db.rs

```diff
@@ -46,9 +49,90 @@ impl PersistentDb {
         PersistentDb {
             db_ref: None,
             cf: None,
+            tx: None,
             memory_db: Some(HashMap::new()),
         }
     }
+
+    /// Commits the pending transactions to the db
+    pub fn flush(&mut self) {
+        // The case when db_ref.is_none and tx.is_some is not a realistic one,
+        // because tx is a DBTransaction created from db_ref
+        if let (Some(db_ref), Some(tx)) = (&self.db_ref, &self.tx) {
```

Comment:
You can use a match statement here. It is much better for readability:
```rust
match (&self.db_ref, &self.tx) {
    (Some(db_ref), Some(tx)) => db_ref.write(tx.clone()).unwrap(),
    (Some(_db_ref), None) => warn!("Unnecessarily called flush before doing any transaction"),
    (None, None) => warn!("Unnecessarily called flush because no db reference can be found")
}
```